test/Transformations_test.jl

```diff
@@ -154,13 +152,32 @@
 
         # Test that the resulting tn contains no tensors with larger rank than the original
         rank = length ∘ size ∘ parent
-        @test max(rank(tensors(reduced)) == max(rank(tensors(tn))))
+        @test max(rank.(tensors(reduced))...) <= max(rank.(tensors(tn))...)
 
         # Test that the resulting tn contains <= tensors than the original
-        @test length(tensors(reduced)) == 1
+        @test length(tensors(reduced)) <= length(tensors(tn))
```

Comment:
Can you replace `length(tensors(...))` for `Tenet.ntensors`?

Reply:
```suggestion
        @test Tenet.ntensors(reduced) <= Tenet.ntensors(tn)
```

Reply:
Why is this better?

Reply:
`tensors` needs to create a `Vector` where it copies all the `Tensor`s while `ntensors` doesn't: it just queries the inner `Dict`

Reply:
@jofrevalles it might be worth checking if that by `@inline`ing `tensors` and `inds`, it can optimize away the `Vector` creation.